presidio-analyzer/presidio_analyzer/predefined_recognizers/in_vehicle_registration_recognizer.py

```diff
@@ -0,0 +1,395 @@
+        "24", "25", "26", "27", "28", "29", "30", "31", "32", "33", "34", "35",
+        "36", "37", "38", "39", "40", "41", "42", "43", "44", "45", "46", "47",
+        "48", "49", "50", "51", "52", "53", "54", "55", "56", "57", "58", "59",
+        "60", "61", "62", "63", "64", "65", "66", "67", "68", "69", "70", "71",
+        "72", "73", "74", "75", "76", "77", "78", "79", "80", "81", "82", "83",
+        "84", "85", "86", "87", "88", "89", "90", "91", "92", "93", "94", "95",
+        "96"
+    ]
+    in_vehicle_dist_wb = [
+        "01", "02", "03", "04", "05", "06", "07", "08", "09", "10", "11", "12",
+        "13", "14", "15", "16", "17", "18", "19", "20", "22", "23", "24", "25",
+        "26", "27", "28", "29", "30", "31", "32", "33", "34", "35", "36", "37",
+        "38", "39", "40", "41", "42", "43", "44", "45", "46", "47", "48", "49",
+        "50", "51", "52", "53", "54", "55", "56", "57", "58", "59", "60", "61",
+        "62", "63", "64", "65", "66", "67", "68", "69", "70", "71", "72", "73",
+        "74", "75", "76", "77", "78", "79", "80", "81", "82", "83", "84", "85",
+        "86", "87", "88", "89", "90", "91", "92", "93", "94", "95", "96", "97",
+        "98"
+    ]
+
+    in_union_territories = ["AN", "CH", "DH", "DL", "JK", "LA", "LD", "PY"]
+    in_old_union_territories = ["CT", "DN"]
+    in_states = [
+        "AP", "AR", "AS", "BR", "CG", "GA", "GJ", "HR", "HP", "JH", "KA", "KL",
+        "MP", "MH", "MN", "ML", "MZ", "NL", "OD", "PB", "RJ", "SK", "TN", "TS",
+        "TR", "UP", "UK", "WB", "UT"
+    ]
+    in_old_states = ["UL", "OR", "UA"]
+    in_non_standard_state_or_ut = ["DD"]
+
+    # fmt: on
+
+    def __init__(
+        self,
+        patterns: Optional[List[Pattern]] = None,
+        context: Optional[List[str]] = None,
+        supported_language: str = "en",
+        supported_entity: str = "IN_VEHICLE_REGISTRATION",
+        replacement_pairs: Optional[List[Tuple[str, str]]] = None,
+    ):
+        self.replacement_pairs = (
+            replacement_pairs
+            if replacement_pairs
+            else [("-", ""), (" ", ""), (":", "")]
+        )
+        patterns = patterns if patterns else self.PATTERNS
+        context = context if context else self.CONTEXT
+        super().__init__(
+            supported_entity=supported_entity,
+            patterns=patterns,
+            context=context,
+            supported_language=supported_language,
+        )
+
+    def validate_result(self, pattern_text: str) -> bool:
+        """Determine absolute value based on calculation."""
+        sanitized_value = Utils.sanitize_value(pattern_text, self.replacement_pairs)
+        # print('Sanitized value:' + sanitized_value)
+        return self.__check_vehicle_registration(sanitized_value)
+
+    def __check_vehicle_registration(self, sanitized_value: str) -> bool:
+        # print('check function called')
+        is_valid_registration = None  # deliberately not typecasted or set to bool False
```

Comment:
Could you please explain the comment? If the method signature says it returns bool, then None should not be one of the options.

Reply:
hi @omri374 , my requirement is that the validate function should kick in for checking 2 of the 9 patterns defined. Now, if it returns a bool : false, then the score is defaulted to 0 (min score) instead of my intended pattern based score as per the logic for this utility function. Hence, I am returning None for the rest of the 7 cases so that I can pick up the pattern based score, instead of 0. Please let me know if there is a better way to implement the same.

Reply:
I see. Ok, let's leave as is. One approach could have been to split to two different recognizers but I think it's probably over engineering.